apiary.apib

```diff
@@ -3281,6 +3281,311 @@ Update the current timer. Only possible if there is a timer running.
 
 + Response 204
 
+# Group Work Orders
+
+## Work Orders [/workOrders]
+
+### workOrders.draft [POST /workOrders.draft]
+
+Add a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + customer (object, required)
+            + type (enum[string], required)
+                + Members
+                    + contact
+                    + company
+            + id: `09e5d75f-f817-4872-9723-57fbb8ff710d` (string, required)
+        + starts_at: `2017-05-26T10:01:49+00:00` (string, required)
+        + description: `Pipework intervention` (string, optional)
+        + department_id: `cef01135-7e51-4f6f-a6eb-6e5e5a885ac7` (string, optional)
+        + layout_id: `a05320f6-b3ef-4ff9-a287-2074a12fb3e9` (string, optional)
+        + milestone_id: `e94c1363-3426-46a4-921d-fce03f39bbd5` (string, optional)
+        + event_id: `9fc14045-5c6c-4ba8-8672-42ea3f26aa63` (string, optional)
+        + location (Address, optional)
+        + distance (Distance, optional)
+        + work_hours (array, optional)
+            + (object)
+                + user_id: `4ec55a8c-2d80-472a-a9c2-5ff5ee7eac6a` (string, required)
+                + started_at: `2017-04-26T10:01:49+00:00` (string, required)
+                + duration (Duration, required)
+                + description (string, optional)
+        + products (array, optional)
+            + (object)
+                + product_id: `e2314517-3cab-4aa9-8471-450e73449040` (string, required)
+                + quantity: `3` (number, required)
+        + custom_fields (array[CustomField], optional)
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (object)
+            + type: `workOrder` (string)
+            + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string)
+
+### workOrders.update [POST /workOrders.update]
+
+Update a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + customer (object, optional)
+            + type (enum[string], required)
+                + Members
+                    + contact
+                    + company
+            + id: `09e5d75f-f817-4872-9723-57fbb8ff710d` (string, required)
+        + starts_at: `2017-05-26T10:01:49+00:00` (string, optional)
+        + description: `Pipework intervention` (string, optional, nullable)
+        + department_id: `cef01135-7e51-4f6f-a6eb-6e5e5a885ac7` (string, optional, nullable)
+        + layout_id: `a05320f6-b3ef-4ff9-a287-2074a12fb3e9` (string, optional, nullable)
```

Comment:
Q: do we want to be able to pass `null` as layout id? I guess that if we draft a work order and we don't pass a layout, the default layout for the current department will be taken, and we don't want to be able to remove that one again?

Reply:
I'd use the same approach here as for the department ID.